backend/battling/views.py

```diff
@@ -1,93 +1,81 @@
 
-class Opponent(TemplateView):  # will become an email
-    template_name = "battling/opponent.html"
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        battle = form.save()
 
+        team = Team.objects.create(battle=battle, trainer=self.request.user)
 
-class BattleEnd(TemplateView):
-    template_name = "battling/battle_end.html"
+        return HttpResponseRedirect(reverse_lazy("create_team", args=(team.id,)))
 
 
-class CreateBattle(CreateView):
-    model = Battle
-    form_class = CreatorForm
-    template_name = "battling/create_battle.html"
-    success_url = reverse_lazy("invite")
+class CreateTeam(UpdateView):
+    model = Team
+    form_class = TeamForm
+    template_name = "battling/create_team.html"
+    success_url = reverse_lazy("home")
 
     def form_valid(self, form):
-        pokemon = form.cleaned_data
+        battle = self.get_object().battle
+        battle_creator = battle.creator
+
+        # self.get_object().trainer will be replaced by self.request.user
+        # when the login feature is implemented
+        if self.get_object().trainer == battle_creator:
+            messages.success(self.request, "Your battle was created!")
 
-        form.instance.pokemon_1 = pokemon["creator_pokemon_1"]
-        form.instance.pokemon_2 = pokemon["creator_pokemon_2"]
-        form.instance.pokemon_3 = pokemon["creator_pokemon_3"]
+        else:
+            messages.success(self.request, "Battle ended! Check e-mail for results.")
 
-        form.instance.save()
+            # run_battle_and_send_email(battle)
 
         return super().form_valid(form)
 
 
-class EnterBattle(UpdateView):
-    model = Battle
-    form_class = OpponentForm
-    template_name = "battling/enter_battle.html"
-    success_url = reverse_lazy("battle_end")
+class JoinBattle(UpdateView):
+    model = Team
+    form_class = CreatorForm
+    template_name = "battling/join_battle.html"
 
     def get_battle(self):
-        id_ = Battle.objects.last().id
-        return get_object_or_404(Battle, id=id_)
+        return get_object_or_404(Battle, id=self.kwargs["pk"])
 
     def form_valid(self, form):
-        pokemon = form.cleaned_data
+        form.instance.creator = self.request.user
+        battle = self.get_battle()
 
-        form.instance.pokemon_1 = pokemon["opponent_pokemon_1"]
-        form.instance.pokemon_2 = pokemon["opponent_pokemon_2"]
-        form.instance.pokemon_3 = pokemon["opponent_pokemon_3"]
+        team = Team.objects.create(battle=battle, trainer=self.request.user)
 
-        form.instance.save()
+        return HttpResponseRedirect(reverse_lazy("create_team", args=(team.id,)))
 
-        form.instance.battle_id = self.get_battle().id
-        run_battle_and_send_email(form.instance.battle_id)
 
-        return super().form_valid(form)
+class DetailBattle(DetailView):
+    template_name = "battling/battle_details.html"
+    model = Battle
 
+    def get_battle(self):
```

Comment:
This should work by default, you can remove the `get_battle` and the `context_object_name`.